File: src/test/java/com/example/JavaStockPriceClient/WebClientStockClientTest.java
package com.example.JavaStockPriceClient;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Flux;

import static org.junit.jupiter.api.Assertions.*;

class WebClientStockClientIntegrationTest {

    private WebClient webClient = WebClient.builder().build();

    @Test
    void retrieveStockPricesFromTheService(){
        WebClientStockClient webClientStockClient = new WebClientStockClient(webClient);
        Flux<StockPrice> prices = webClientStockClient.pricesFor("SYMBOL");

        Assertions.assertNotNull(prices);
        Flux<StockPrice> fivePrices = prices.take(5);
        Assertions.assertEquals(5, fivePrices.count().block());
        Assertions.assertEquals("SYMBOL", fivePrices.blockFirst().getSymbol());

    }
}

File: src/test/kotlin/se/newton/stockpriceclient/WebClientStockClientTest.kt
package se.newton.stockpriceclient

import org.springframework.web.reactive.function.client.WebClient
import kotlin.test.assertEquals
import kotlin.test.Test

class WebClientStockClientTest {
  private val webClient = WebClient.builder().build()

  @Test
  fun `retrieve stock prices from the service`() {
		// Arrange
    val webClientStockClient = WebClientStockClient(webClient)

		// Act
    val prices = webClientStockClient.pricesFor("SYMBOL")
    val fivePrices = prices.take(5)

		// Assert
    assertEquals(5, fivePrices.count().block())
		assertEquals("SYMBOL", fivePrices.blockFirst()?.symbol)
  }
}